lib/hostinfo.cpp

```diff
@@ -291,3 +312,158 @@ int HOST_INFO::write_cpu_benchmarks(FILE* out) {
     );
     return 0;
 }
+
+bool HOST_INFO::parse_linux_os_info(FILE* file, const LINUX_OS_INFO_PARSER parser, 
```

Comment:
I'm not sure this belongs in this file. The file is about parsing our own XML file into a structure and is platform neutral. The `parse_linux_os_info(...)` stuff is linux specific and has nothing to do with the XML file. The idea is valid though and this should either stay in `hostinfo_unix.cpp` or go into a new `hostinfo_linux.cpp`.

Reply:
But here's the idea to reuse linux parsing code in windows to parse linux :)

Reply:
Fixed